File: Problem 3/C++/problem3.cpp
#include <iostream>
#include <math.h>

bool isPrime(int k)
{
	int factors = 0;
	for(int i = 1; i < sqrt(k)+1; ++i)
	{
		if(k%i == 0)
		{
			++factors;
		}

		if(factors >= 2)
		{
			return false;
		}
	}
	return true;
}

int main()
{
	int factor = 3;
	int pFactor = 0;
	unsigned long long int number = 600851475143;

	while(factor < sqrt(number)+1)
	{
		if(number % factor == 0 && factor % 2 != 0)
		{
			if(isPrime(factor))
			{
				pFactor = factor;
			}
		}

		factor+=2;
	}
	std::cout <<pFactor << std::endl;

	return 0;
}

File: Problem 3/Nim/problem3.nim
import math

#Determine if a given number is prime
proc isPrime(n: int64): bool =
    var factors: int = 0
    if n mod 2 == 0 and n != 2:
        return false
    for i in 1..int (sqrt(float n) + 1):
        if n mod i == 0:
            factors += 1
    return factors < 2

#Return the largest prime factor of n
proc problem3(n: int64): int =
    var largestPrimeFactor: int = 3
    for i in countup(3, int sqrt(float n) + 1, 2):
        if n mod i == 0 and isPrime(i):
            if i > largestPrimeFactor:
                largestPrimeFactor = i
    return largestPrimeFactor


echo(problem3(600851475143))
